src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -344,7 +344,8 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
     DatabasePtr database;
     {
         std::lock_guard lock{databases_mutex};
-        auto it = databases.find(table_id.getDatabaseName());
+        // hasDatabase() to avod getDatabaseName() throwing exception if database is empty.
```

Comment:
The comment is not explanatory enough. It does not explain the motivation for this logic.